### PR TITLE
Fixed path to the CA cert.

### PR DESCRIPTION
**Pull request by jpkrohling**

### CODE REVIEW

**Comment by jpkrohling** [comment]
Without this PR:

```
I0419 16:18:38.627329       1 hawkular-openshift-agent.go:69] Hawkular OpenShift Agent: Version: 1.4.2.Final-SNAPSHOT, Commit: 650038606decb4a7e42570bc51715121cf3e694f
I0419 16:18:38.628010       1 hawkular-openshift-agent.go:70] DEBUG: Hawkular OpenShift Agent Command line: [/opt/hawkular/hawkular-openshift-agent -config /hawkular-openshift-agent-configuration/config.yaml -v 4]
I0419 16:18:38.628020       1 config.go:275] DEBUG: Reading YAML config from [/hawkular-openshift-agent-configuration/config.yaml]
I0419 16:18:38.628639       1 emitter_server.go:48] Agent emitter will emit metrics
I0419 16:18:38.628658       1 emitter_server.go:62] Agent emitter will emit status
I0419 16:18:38.628666       1 emitter_server.go:70] Agent emitter will provide a health probe
I0419 16:18:38.628673       1 emitter_server.go:96] Agent will start the emitter endpoint at [:8080]
E0419 16:18:38.683512       1 metrics_storage.go:201] Error trying to get Secret named [hawkular-metrics-certificate] from namespace [openshift-infra]: err=secrets "hawkular-metrics-certificate" not found
E0419 16:18:38.683549       1 metrics_storage.go:217] Could not get the requested secret. This may mean the secret has not yet been created yet or the agent does not have the required permission to access the secret. Will attempt again every 5 seconds for the next 5 minutes
E0419 16:18:43.691842       1 metrics_storage.go:201] Error trying to get Secret named [hawkular-metrics-certificate] from namespace [openshift-infra]: err=secrets "hawkular-metrics-certificate" not found
E0419 16:18:48.700153       1 metrics_storage.go:201] Error trying to get Secret named [hawkular-metrics-certificate] from namespace [openshift-infra]: err=secrets "hawkular-metrics-certificate" not found
```

With this PR:
```
I0419 16:16:11.076776       1 hawkular-openshift-agent.go:69] Hawkular OpenShift Agent: Version: 1.4.2.Final-SNAPSHOT, Commit: 650038606decb4a7e42570bc51715121cf3e694f
I0419 16:16:11.077494       1 hawkular-openshift-agent.go:70] DEBUG: Hawkular OpenShift Agent Command line: [/opt/hawkular/hawkular-openshift-agent -config /hawkular-openshift-agent-configuration/config.yaml -v 4]
I0419 16:16:11.077503       1 config.go:275] DEBUG: Reading YAML config from [/hawkular-openshift-agent-configuration/config.yaml]
I0419 16:16:11.080715       1 emitter_server.go:48] Agent emitter will emit metrics
I0419 16:16:11.080744       1 emitter_server.go:62] Agent emitter will emit status
I0419 16:16:11.080767       1 emitter_server.go:70] Agent emitter will provide a health probe
I0419 16:16:11.080785       1 emitter_server.go:96] Agent will start the emitter endpoint at [:8080]
I0419 16:16:11.157826       1 metrics_storage.go:75] START storing metrics definitions and data
I0419 16:16:11.157852       1 metrics_collector_manager.go:66] Pods will be limited to a maximum of [50] metrics
I0419 16:16:11.157859       1 metrics_collector_manager.go:67] Endpoints will be limited to a minimum collection interval of [10s]
I0419 16:16:11.165118       1 get_local_node.go:50] DEBUG: Agent pod [default/hawkular-openshift-agent-m6r1k] has node name of [localhost]
I0419 16:16:11.171045       1 discovery.go:114] DEBUG: Starting Kubernetes Discovery
I0419 16:16:11.216610       1 discovery.go:267] DEBUG: Detected a new pod that was added: localhost/openshift-infra/hawkular-metrics-912ph/a5a74684-2517-11e7-9a66-c85b7644b6b5
I0419 16:16:11.216642       1 discovery.go:145] DEBUG: Changed pod [localhost/openshift-infra/hawkular-metrics-912ph/a5a74684-2517-11e7-9a66-c85b7644b6b5] does not have volume [hawkular-openshift-agent]
I0419 16:16:11.223425       1 discovery.go:267] DEBUG: Detected a new pod that was added: localhost/default/hawkular-openshift-agent-m6r1k/7f413eba-251b-11e7-9a66-c85b7644b6b5
I0419 16:16:11.223460       1 discovery.go:141] DEBUG: Changed pod [localhost/default/hawkular-openshift-agent-m6r1k/7f413eba-251b-11e7-9a66-c85b7644b6b5] with volume [hawkular-openshift-agent=hawkular-openshift-agent-configuration] does not have a config map
I0419 16:16:11.238092       1 discovery.go:267] DEBUG: Detected a new pod that was added: localhost/openshift-infra/metrics-deployer-pod-thtcj/93145ae4-2517-11e7-9a66-c85b7644b6b5
I0419 16:16:11.238131       1 discovery.go:145] DEBUG: Changed pod [localhost/openshift-infra/metrics-deployer-pod-thtcj/93145ae4-2517-11e7-9a66-c85b7644b6b5] does not have volume [hawkular-openshift-agent]
I0419 16:16:11.242733       1 discovery.go:368] DEBUG: Detected a new configmap [hawkular-openshift-agent-configuration] for namespace [default]
I0419 16:16:11.242926       1 discovery.go:172] DEBUG: Changed configmap [hawkular-openshift-agent-configuration] for namespace [default] affects pod [localhost/default/hawkular-openshift-agent-m6r1k/7f413eba-251b-11e7-9a66-c85b7644b6b5] with volume: [hawkular-openshift-agent=hawkular-openshift-agent-configuration]
W0419 16:16:11.242985       1 node_event_consumer.go:169] Will not start collecting for endpoint in pod [localhost/default/hawkular-openshift-agent-m6r1k/7f413eba-251b-11e7-9a66-c85b7644b6b5] - cannot build URL. err=Endpoint not assigned an IP address yet
I0419 16:16:11.243285       1 discovery.go:267] DEBUG: Detected a new pod that was added: localhost/default/persistent-volume-setup-w8v6s/9350a971-2517-11e7-9a66-c85b7644b6b5
I0419 16:16:11.243310       1 discovery.go:145] DEBUG: Changed pod [localhost/default/persistent-volume-setup-w8v6s/9350a971-2517-11e7-9a66-c85b7644b6b5] does not have volume [hawkular-openshift-agent]
I0419 16:16:11.253481       1 discovery.go:267] DEBUG: Detected a new pod that was added: localhost/default/docker-registry-1-q9hp5/9de84a2f-2517-11e7-9a66-c85b7644b6b5
I0419 16:16:11.253572       1 discovery.go:145] DEBUG: Changed pod [localhost/default/docker-registry-1-q9hp5/9de84a2f-2517-11e7-9a66-c85b7644b6b5] does not have volume [hawkular-openshift-agent]
I0419 16:16:11.259779       1 discovery.go:267] DEBUG: Detected a new pod that was added: localhost/default/router-1-hd3bf/9e30d432-2517-11e7-9a66-c85b7644b6b5
I0419 16:16:11.259809       1 discovery.go:145] DEBUG: Changed pod [localhost/default/router-1-hd3bf/9e30d432-2517-11e7-9a66-c85b7644b6b5] does not have volume [hawkular-openshift-agent]
I0419 16:16:11.266023       1 discovery.go:267] DEBUG: Detected a new pod that was added: localhost/openshift-infra/hawkular-cassandra-1-tsc4v/a5bf95a3-2517-11e7-9a66-c85b7644b6b5
I0419 16:16:11.266052       1 discovery.go:145] DEBUG: Changed pod [localhost/openshift-infra/hawkular-cassandra-1-tsc4v/a5bf95a3-2517-11e7-9a66-c85b7644b6b5] does not have volume [hawkular-openshift-agent]
I0419 16:16:11.270998       1 discovery.go:267] DEBUG: Detected a new pod that was added: localhost/openshift-infra/heapster-bmmg5/a6f54b69-2517-11e7-9a66-c85b7644b6b5
I0419 16:16:11.271048       1 discovery.go:145] DEBUG: Changed pod [localhost/openshift-infra/heapster-bmmg5/a6f54b69-2517-11e7-9a66-c85b7644b6b5] does not have volume [hawkular-openshift-agent]
I0419 16:16:11.595671       1 discovery.go:298] DEBUG: Detected a modified pod: localhost/default/hawkular-openshift-agent-m6r1k/7f413eba-251b-11e7-9a66-c85b7644b6b5
I0419 16:16:11.595719       1 discovery.go:138] DEBUG: Changed pod [localhost/default/hawkular-openshift-agent-m6r1k/7f413eba-251b-11e7-9a66-c85b7644b6b5] with volume [hawkular-openshift-agent=hawkular-openshift-agent-configuration] has a config map
I0419 16:16:11.596014       1 metrics_collector_manager.go:140] START collecting metrics from [default|hawkular-openshift-agent-m6r1k|prometheus|http://172.17.0.3:8080/metrics] every [30s]
I0419 16:16:12.603791       1 discovery.go:298] DEBUG: Detected a modified pod: localhost/default/hawkular-openshift-agent-m6r1k/7f413eba-251b-11e7-9a66-c85b7644b6b5
I0419 16:16:12.603829       1 discovery.go:138] DEBUG: Changed pod [localhost/default/hawkular-openshift-agent-m6r1k/7f413eba-251b-11e7-9a66-c85b7644b6b5] with volume [hawkular-openshift-agent=hawkular-openshift-agent-configuration] has a config map
I0419 16:16:12.603874       1 metrics_collector_manager.go:386] STOP collecting metrics from [default|hawkular-openshift-agent-m6r1k|prometheus|http://172.17.0.3:8080/metrics]
I0419 16:16:12.604023       1 metrics_collector_manager.go:140] START collecting metrics from [default|hawkular-openshift-agent-m6r1k|prometheus|http://172.17.0.3:8080/metrics] every [30s]
I0419 16:16:42.604268       1 prometheus_metrics_collector.go:97] DEBUG: Told to collect all Prometheus metrics from [http://172.17.0.3:8080/metrics]
I0419 16:16:42.613804       1 prometheus_metrics_collector.go:235] DEBUG: Told to collect details on [35] Prometheus metrics from [http://172.17.0.3:8080/metrics]
I0419 16:16:42.716660       1 metrics_storage.go:167] DEBUG: Stored datapoints for [35] metrics
```

**Comment by jpkrohling** [comment]
Initially, I thought the service CA from OpenShift would be enough to trust the certificate, but it's not: accessing metrics via its cluster address causes the cert created by `metrics-signer` to be served, so, we need to mount that secret for now.

```
curl -v https://hawkular-metrics.openshift-infra.svc.cluster.local
* About to connect() to hawkular-metrics.openshift-infra.svc.cluster.local port 443 (#0)
*   Trying 172.30.145.243...
* Connected to hawkular-metrics.openshift-infra.svc.cluster.local (172.30.145.243) port 443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
* Server certificate:
* 	subject: CN=hawkular-metrics
* 	start date: Apr 19 15:48:26 2017 GMT
* 	expire date: Apr 19 15:48:27 2019 GMT
* 	common name: hawkular-metrics
* 	issuer: CN=metrics-signer@1492616905
```

**Comment by jmazzitelli** [comment]
Note that merging this will mean HOSA using this default config will break when running with older versions of Origin Metrics. Going to wait to merge this. Perhaps we can figure someway to support the latest Origin Metrics while also supporting older versions. Perhaps use env vars in the Makefile? The agent supports the environment variable HAWKULAR_SERVER_CA_CERT_FILE.

File: deploy/openshift/hawkular-openshift-agent-configmap.yaml [hunk 1]
```diff
@@ -11,11 +11,11 @@ data:
     kubernetes:
       tenant: ${POD:namespace_name}
     hawkular_server:
-      url: https://hawkular-metrics.openshift-infra.svc.cluster.local       
+      url: https://hawkular-metrics.openshift-infra.svc.cluster.local
       credentials:
         username: secret:openshift-infra/hawkular-metrics-account/hawkular-metrics.username
         password: secret:openshift-infra/hawkular-metrics-account/hawkular-metrics.password
-      ca_cert_file: secret:openshift-infra/hawkular-metrics-certificate/hawkular-metrics-ca.certificate
+      ca_cert_file: secret:openshift-infra/hawkular-metrics-certs/tls.truststore.crt
```
Comment (mwringe) [inline deploy/openshift/hawkular-openshift-agent-configmap.yaml]: This doesn't seem right, why are use using the tls.truststore.crt here?

Comment (jpkrohling) [inline deploy/openshift/hawkular-openshift-agent-configmap.yaml]: > Initially, I thought the service CA from OpenShift would be enough to trust the certificate, but it's not: accessing metrics via its cluster address causes the cert created by metrics-signer to be served, so, we need to mount that secret for now.

Comment (jpkrohling) [inline deploy/openshift/hawkular-openshift-agent-configmap.yaml]: Either we use that, or we build HOSA's own trust store at a separate secret. Using the system's CA isn't enough, as there's an intermediate CA that is unknown to the caller.

Comment (steveteuber) [inline deploy/openshift/hawkular-openshift-agent-configmap.yaml]: The heapster pod in the `openshift-infra` project use secret `hawkular-metrics-certs/tls.crt` as CA file.
```
$ oc get rc heapster -n openshift-infra -o yaml | grep hawkular-metrics-certs/
--sink=hawkular:https://hawkular-metrics:443 ... &caCert=/hawkular-metrics-certs/tls.crt
```

Can you update the pull request to use this line?
```
ca_cert_file: secret:openshift-infra/hawkular-metrics-certs/tls.crt
```

**Comment by slaskawi** [comment]
Could you please tell me what's the ETA for this Pull Request? I still see those errors when running OpenShift Origin v3.6.0-rc.0 with `oc cluster up --metrics=true`.

**Comment by jpkrohling** [comment]
PR updated with the suggestion from @steveteuber

**Pull request merged**